ocaml/xapi/xapi_host.ml

```diff
@@ -2652,6 +2652,8 @@ let alert_if_tls_verification_was_emergency_disabled ~__context =
 let cert_distrib_atom ~__context ~host ~command =
   Cert_distrib.local_exec ~__context ~command
 
+let copy_primary_host_certs ~__context ~host = ()
```

Comment:
Can we add a comment this is added to appease the type system?

Reply:
Or move the actual business logic (`Cert_distrib.copy_certs_to_host ~__context ~host`) here, which would be the usual place (leaving message_forwarding for matter related to handling pool operations and forwarding).

Reply:
it's important that the master is the host coordinating the cert copy, because otherwise the cert distrib lock would not be taken. so the options are:

 - slave forwards copy request to master via message forwarding (as currently implemented)
 - we perform the forwarding inside cert_distrib.ml (is this what you were suggesting implicitly @robhoes  ...?)

Reply:
No, I mean that still `copy_primary_host_certs` is called through the API, with the implementation right here in xapi_host.ml. To make sure that the function runs on the master only, message_forwarding would only have a call to `Local.Host. copy_primary_host_certs`, no forwarding to a slave with `Client.Host.copy_primary_host_certs` through `do_op_on`. Slaves call the function only through `Client.Host.copy_primary_host_certs` and `Helpers.call_api_functions`, which go to the master.

Reply:
yes this is the same as what I have done, modulo some copy pasting (which I will do now)